markdown/guide/testing.md

```diff
@@ -1,3 +1,160 @@
 ## Testing Engines
 
-_TODO: PRs welcome! We haven't quite nailed down the testing story for Engines and especially Lazy Engines, so if you have ideas reach out!_
+> The in-repo-engine & in-repo-addon don't have support for separate tests. If you want to create a separation of concerns between the app and your engine, extract the functionality into a normal addon.
+
+* To test an `in-repo-engine`, we must set up the tests from the host app
+
+* To test an `engine as an addon`, we must set up the tests from the dummy app
+
+Ember Engines comes with a **set of test helpers that can be used in Unit/Integration tests**:
+
+`engineResolverFor` gets the resolver class used by an Engine and creates an instance to be used with test modules:
+
+```js
+// tests/<integration or unit>/…
+import engineResolverFor from 'ember-engines/test-support/engine-resolver-for';
+```
+
+### Unit/Integration Testing
+
+
+To test candidates for unit/integration (e.g. components, services and controllers) declared inside an engine, you need to set a custom resolver with the engine's prefix using `engineResolverFor` helper.
+
+What does it look like to test a component from a host app or dummy app? Let's go over some examples in the next section. In the following tests, `admin-app` is an `in-repo-engine` or `engine as addon`, `hello-name` is a component, and `some-thing` is a service.
```

Comment:
I think this is very confusing to call our engine `admin-app`, which makes it sound like an "app". How about `admin-console` or `admin-section` or even `admin-engine`?

Reply:
Also, let's just use a single term `in-repo-engine` - saying `or 'engine as addon'` is super confusing.

Reply:
@dgeb how can we referrer here? `in-repo-engine` and `addon-engine`?

Reply:
> `in-repo-engine` and `addon-engine`

Sure that's fine to distinguish between the two flavors of engines

My objection is that this phrase doesn't make sense:

> `admin-app` is an `in-repo-engine` or `engine as addon`

It has to be one or the other.

Reply:
fixed!